apps/condo/domains/contact/utils/helpers.spec.ts

```diff
@@ -0,0 +1,266 @@
+                } },
+            ]
+            describe('should generate correct filters', () => {
+                describe('if 1 filter defined', () => {
+                    for (let i = 0; i < filters.length; i++) {
+                        it(`${filters[i].key}`, () => {
+                            const testFilters = {}
+                            testFilters[filters[i].key] = filters[i].key
+                            expect(filtersToQuery(testFilters)).toStrictEqual({
+                                AND: [
+                                    filters[i].result,
+                                ],
+                            })
+                        })
+                    }
+                })
+                describe('if 2 filters defined', () => {
+                    for (let i = 0; i < filters.length - 1; i++) {
+                        for (let j = i + 1; j < filters.length; j++) {
+                            it(`${filters[i].key} + ${filters[j].key}`, () => {
+                                const testFilters = {}
+                                testFilters[filters[i].key] = filters[i].key
+                                testFilters[filters[j].key] = filters[j].key
+                                expect(filtersToQuery(testFilters)).toStrictEqual({
+                                    AND: [
+                                        filters[i].result,
+                                        filters[j].result,
+                                    ],
+                                })
+                            })
+                        }
+                    }
+                })
+                describe('if 3 filters defined', () => {
+                    for (let i = 0; i < filters.length - 2; i++) {
+                        for (let j = i + 1; j < filters.length - 1; j++) {
+                            for (let k = j + 1; k < filters.length; k++) {
+                                it(`${filters[i].key} + ${filters[j].key}`, () => {
+                                    const testFilters = {}
+                                    testFilters[filters[i].key] = filters[i].key
+                                    testFilters[filters[j].key] = filters[j].key
+                                    testFilters[filters[k].key] = filters[k].key
+                                    expect(filtersToQuery(testFilters)).toStrictEqual({
+                                        AND: [
+                                            filters[i].result,
+                                            filters[j].result,
+                                            filters[k].result,
+                                        ],
+                                    })
+                                })
+                            }
+                        }
+                    }
+                })
+                describe('if 4 filters defined', () => {
+                    for (let i = 0; i < filters.length - 3; i++) {
```

Comment:
Looks strange

Reply:
This set of tests determines possibly of applying all filters in any combinations